Assets/MRTK/SDK/Features/DataBinding/Scripts/DataConsumers/DataConsumerImageTextureFromUrl.cs

```diff
@@ -0,0 +1,151 @@
+        internal int _frameDelayCountdown = 0;
+        internal System.Random _random;
+        internal Material _imageMaterial;
+
+        internal readonly float FramesPerMillisecond = 60.0f / 1000.0f; // 60 frames per 1000 milliscends
+
+        void Update()
+        {
+            CheckForWaitingUrlToFetch();
+
+        }
+
+        internal override bool ManageChildren()
+        {
+            return manageChildren;
+        }
+
+        internal override void InitializeDataConsumer()
+        {
+            _random = new System.Random();
+        }
+
+        internal override void AddVariableKeyPathsForComponent(Type componentType, Component component)
+        {
+            InitializeForComponent(componentType, component);
+
+            AddKeyPath(this.keyPath);
+        }
+
+
+        internal abstract void InitializeForComponent(Type componentType, Component component);
+        internal abstract void PlaceImageTexture(Texture2D imageTexture);
+
+
+        internal override void ProcessDataChanged(IDataSource dataSource, string resolvedKeyPath, string localKeyPath, object newValue)
+        {
+            // Debug.Log("ProcessDataChanged " + resolvedKeyPath + ",  local: " + localKeyPath);
+
+            if (localKeyPath == this.keyPath)
+            {
+                string newUrl = newValue.ToString();
+                _frameDelayCountdown = (int)((float)maxRandomLoadBalancingDelayInMilliseconds * FramesPerMillisecond);
+                _frameDelayCountdown = _random.Next(0, _frameDelayCountdown);
+
+                if (newUrl != null)
+                {
+                    if (_fetchInProgress || _frameDelayCountdown > 0)
+                    {
+                        _waitingUrlToFetch = newUrl;
+                    }
+                    else
+                    {
+                        _fetchInProgress = true;
+
+                        StartCoroutine(FetchImageTexture(newUrl));
+                    }
+                }
+            }
+        }
+
+        internal void CheckForWaitingUrlToFetch()
+        {
+            if (_waitingUrlToFetch != null && _fetchInProgress == false && --_frameDelayCountdown <= 0)
+            {
+                _fetchInProgress = true;
+                StartCoroutine(FetchImageTexture(_waitingUrlToFetch));
+                _waitingUrlToFetch = null;
+            }
+        }
+
+
+        internal IEnumerator FetchImageTexture(string uri)
+        {
+            bool doneWithSuccess = false;
+            const int kMaxRetries = 3;
+
+            // UnityEngine.Debug.Log("SendWebRequest: " + uri);
+            for (int retries = 0; !doneWithSuccess && retries < kMaxRetries; retries++)
+            {
+                using (UnityWebRequest webRequest = UnityWebRequestTexture.GetTexture(uri))
+                {
+                    yield return webRequest.SendWebRequest();
+
+                    if (webRequest.isHttpError || webRequest.isNetworkError)
+                    {
```

Comment:
`webRequest.isHttpError` and `webRequest.isNetworkError` are obsolete starting in Unity 2020.2

https://docs.unity3d.com/2020.2/Documentation/ScriptReference/Networking.UnityWebRequest-isNetworkError.html

```
#if UNITY_2020_2_OR_NEWER
                    if (webRequest.result == UnityWebRequest.Result.ProtocolError || webRequest.result == UnityWebRequest.Result.ConnectionError)
                    {
#else
                    if (webRequest.isHttpError || webRequest.isNetworkError)
                    {
#endif
```